ibis/backends/dask/execution/window.py

```diff
@@ -84,3 +99,67 @@ def execute_window_op(
         **kwargs,
     )
     return _post_process_empty(result, data, timecontext)
+
+
+def execute_grouped_window_op(
+    op, data, window, scope, timecontext, aggcontext, clients, **kwargs,
+):
+    # extract the parent
+    (root,) = op.root_tables()
+    root_expr = root.to_expr()
+
+    data = execute(
+        root_expr,
+        scope=scope,
+        timecontext=timecontext,
+        clients=clients,
+        aggcontext=aggcontext,
+        **kwargs,
+    )
+
+    group_by = window._group_by
+    grouping_keys = [
+        key_op.name
+        if isinstance(key_op, ops.TableColumn)
+        else execute(
+            key,
+            scope=scope,
+            clients=clients,
+            timecontext=timecontext,
+            aggcontext=aggcontext,
+            **kwargs,
+        )
+        for key, key_op in zip(
+            group_by, map(operator.methodcaller('op'), group_by)
+        )
+    ]
+
+    grouped_root_data = data.groupby(grouping_keys)
+    scope = scope.merge_scopes(
+        [
+            Scope({t: grouped_root_data}, timecontext)
+            for t in op.expr.op().root_tables()
+        ],
+        overwrite=True,
+    )
+
+    result = execute_with_scope(
+        expr=op.expr,
+        scope=scope,
+        timecontext=timecontext,
+        aggcontext=aggcontext,
+        clients=clients,
+        **kwargs,
+    )
+    # If the grouped operation we performed is not an analytic UDF we have to
+    # realign the output to the input.
+    if not isinstance(op.expr._arg, ops.AnalyticVectorizedUDF):
+        result = dd.merge(
```

Comment:
This feels a bit suboptimal because when you have many udfs inside (people often compute many window summarization for different columns at once), you would ended up doing many merging. 

Not an issue for this PR but I suggest leaving a note here to look into better ways of doing this in the future.

Reply:
Created https://github.com/ibis-project/ibis/issues/2880 as a followup

Reply:
kk thanks